pkg/splunk_logger/splunk_logger.go

```diff
@@ -0,0 +1,147 @@
+package logger
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+const (
+	PayloadsChannelSize = 1000
+	// in seconds, how often batched events should be sent
+	SendFrequency = 5
+)
+
+type SplunkLogger struct {
+	client   *http.Client
+	url      string
+	token    string
+	source   string
+	hostname string
+
+	payloads chan *SplunkPayload
+}
+
+type SplunkPayload struct {
+	// splunk expects unix time in seconds
+	Time  int64       `json:"time"`
+	Host  string      `json:"host"`
+	Event SplunkEvent `json:"event"`
+}
+
+type SplunkEvent struct {
+	Message string `json:"message"`
+	Ident   string `json:"ident"`
+	Host    string `json:"host"`
+}
+
+func NewSplunkLogger(url, token, source, hostname string) *SplunkLogger {
+	sl := &SplunkLogger{
+		client:   retryablehttp.NewClient().StandardClient(),
+		url:      url,
+		token:    token,
+		source:   source,
+		hostname: hostname,
+	}
+
+	ticker := time.NewTicker(time.Second * SendFrequency)
+	sl.payloads = make(chan *SplunkPayload, PayloadsChannelSize)
+
+	go sl.flushPayloads(ticker.C)
```

Comment:
This line creates a potentially dangling goroutine (there's no way to return from `flushPayloads`). This is usually not an issue, but it may create small annoyances in certain cases (tests).

I suggest passing a `context.Context` into `NewSplunkLogger` and then to `flushPayloads`. When the context gets cancelled, make the function return (possibly after calling `SendPayloads` for one last time?).